1436790/1438462: Part 9: 24 files with MSDN link fixes.

DennisLee-DennisLee opened the pull request:
Contributes to PR #[2037](https://github.com/dotnet/docs/issues/2037) and applies to the following work items:
- [User Story 1436790: Content Update: update MSDN links for .NET content](https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1436790)
- [Task 1438462: Update MSDN links for .NET content](https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1438462)

DennisLee-DennisLee committed:
50 files with MSDN link fixes.

DennisLee-DennisLee committed:
50 files with MSDN link fixes.

DennisLee-DennisLee committed:
Merge branch 'master' of https://github.com/dotnet/docs into v-dele-1438462-009

DennisLee-DennisLee committed:
Revert "50 files with MSDN link fixes."

This reverts commit 200ab21bdc247200750812e12967c0cc8f0ab2ac.

DennisLee-DennisLee committed:
Revert "50 files with MSDN link fixes."

This reverts commit 4918685f3ab9c441718d90f78ca2987844676fc7.

DennisLee-DennisLee committed:
20 more files with MSDN link fixes.

(Cherry-picked from commit 200ab21bdc247200750812e12967c0cc8f0ab2ac, which had 50 file changes.)

DennisLee-DennisLee committed:
Merge branch 'master' of https://github.com/dotnet/docs into v-dele-1438462-009

mairaw commented on docs/framework/data/adonet/ef/language-reference/linq-to-entities.md:
```diff
@@ -84,6 +84,6 @@ LINQ to Entities provides Language-Integrated Query (LINQ) support that enables
 ## See also
 - [Known Issues and Considerations in LINQ to Entities](../../../../../../docs/framework/data/adonet/ef/language-reference/known-issues-and-considerations-in-linq-to-entities.md)
-- [LINQ (Language-Integrated Query)](https://msdn.microsoft.com/library/a73c4aec-5d15-4e98-b962-1274021ea93d)
+- [LINQ (Language-Integrated Query)](https://docs.microsoft.com/previous-versions/visualstudio/visual-studio-2013/bb397926(v=vs.120))
```
There are 13 topics that use this link:
https://github.com/dotnet/docs/search?q=a73c4aec-5d15-4e98-b962-1274021ea93d&unscoped_q=a73c4aec-5d15-4e98-b962-1274021ea93d

They should all be changed to something similar to
```suggestion
- [Language-Integrated Query (LINQ) - C#](../../../../../csharp/programming-guide/concepts/linq/index.md)
- [Language-Integrated Query (LINQ) - Visual Basic](../../../../../visual-basic/programming-guide/concepts/linq/index.md)
```

mairaw reviewed (changes requested):
Thanks @DennisLee-DennisLee! Left a few comments for you to take care of before we can merge,

DennisLee-DennisLee commented on docs/framework/data/adonet/ef/language-reference/add.md:
```diff
@@ -33,4 +33,4 @@ expression + expression
 ## See also
 - [Entity SQL Reference](../../../../../../docs/framework/data/adonet/ef/language-reference/entity-sql-reference.md)
-- [Conceptual Model Types (CSDL)](https://msdn.microsoft.com/library/987b995f-e429-4569-9559-b4146744def4)
+- [Conceptual Model Types (CSDL)](https://docs.microsoft.com/previous-versions/dotnet/netframework-4.0/bb399548(v=vs.100))
```
I've fixed the other common links for CSDL and LINQ in PR #10330 .

mairaw committed:
Apply suggestions from code review

Accepted the 3 suggested fixes from Maira.

Co-Authored-By: DennisLee-DennisLee <v-dele@microsoft.com>

DennisLee-DennisLee committed:
Merge branch 'master' of https://github.com/DennisLee-DennisLee/docs into v-dele-1438462-009

DennisLee-DennisLee committed:
Addressing Maira's request to redirect a couple of pages.

The csdl-ssdl-and-msl-specifications article is a node parent, so I ended up deleting and redirecting four pages in total.

DennisLee-DennisLee commented on docs/framework/data/adonet/ef/language-reference/csdl-ssdl-and-msl-specifications.md:
```diff
@@ -4,6 +4,5 @@ ms.date: "03/30/2017"
 ms.assetid: bbdc9237-ff4c-4441-9565-31ebc29743e9
 ---
 # CSDL, SSDL, and MSL Specifications
```
I redirected to https://docs.microsoft.com/ef/ef6/index, since that's what the MSDN link pointed to. There are three different possibilities for redirect targets, so I don't know if there's a better way to handle it.

This file was a TOC parent to three other pages (including the other one you wanted to delete), so I deleted and set up redirects for four articles in total.

DennisLee-DennisLee committed:
Fixed the AdventureWorks links.

mairaw commented on .openpublishing.redirection.json:
```diff
@@ -2232,6 +2232,26 @@
             "redirect_url": "/visualstudio/vb6/vb6-support",
             "redirect_document_id": true
+        },
+        {
```
let's move this block to line 768 since we try to keep things in alphabetical order

mairaw commented on .openpublishing.redirection.json:
```diff
@@ -2232,6 +2232,26 @@
+        },
+        {
+            "source_path": "docs/framework/data/adonet/ef/language-reference/csdl-ssdl-and-msl-specifications.md",
+            "redirect_url": "/ef/ef6/index",
```
I'd also move to this one since you can find the other two in here as well

```suggestion
            "redirect_url": "/ef/ef6/modeling/designer/advanced/edmx/csdl-spec",
```

mairaw reviewed (changes requested):
A few last suggestion for the redirection file and we're good to go

DennisLee-DennisLee committed:
Addressed Maira's two redirection comments.

DennisLee-DennisLee committed:
Addressed Maira's last comment in part 10 about the "Navigation Properties" link.

mairaw commented on .openpublishing.redirection.json:
```diff
@@ -765,6 +765,26 @@
         },
+        {
+            "source_path": "docs/framework/data/adonet/ef/language-reference/csdl-specification.md",
+            "redirect_url": "/ef/ef6/modeling/designer/advanced/edmx/csdl-spec",
```
```suggestion
            "redirect_url": "/ef/ef6/modeling/designer/advanced/edmx/csdl-spec"
```

mairaw commented on .openpublishing.redirection.json:
```diff
@@ -765,6 +765,26 @@
+        {
+            "source_path": "docs/framework/data/adonet/ef/language-reference/csdl-specification.md",
+            "redirect_url": "/ef/ef6/modeling/designer/advanced/edmx/csdl-spec",
+            "redirect_document_id": true
```
```suggestion
```

mairaw commented on .openpublishing.redirection.json:
```diff
@@ -765,6 +765,26 @@
+        {
+            "source_path": "docs/framework/data/adonet/ef/language-reference/csdl-ssdl-and-msl-specifications.md",
+            "redirect_url": "/ef/ef6/modeling/designer/advanced/edmx/csdl-spec",
+            "redirect_document_id": true
```
```suggestion
```

mairaw commented on .openpublishing.redirection.json:
```diff
@@ -765,6 +765,26 @@
+        },
+        {
+            "source_path": "docs/framework/data/adonet/ef/language-reference/csdl-ssdl-and-msl-specifications.md",
+            "redirect_url": "/ef/ef6/modeling/designer/advanced/edmx/csdl-spec",
```
```suggestion
            "redirect_url": "/ef/ef6/modeling/designer/advanced/edmx/csdl-spec"
```

mairaw commented on .openpublishing.redirection.json:
```diff
@@ -765,6 +765,26 @@
+        {
+            "source_path": "docs/framework/data/adonet/ef/language-reference/msl-specification.md",
+            "redirect_url": "/ef/ef6/modeling/designer/advanced/edmx/msl-spec",
+            "redirect_document_id": true
```
```suggestion
```

mairaw commented on .openpublishing.redirection.json:
```diff
@@ -765,6 +765,26 @@
+        },
+        {
+            "source_path": "docs/framework/data/adonet/ef/language-reference/msl-specification.md",
+            "redirect_url": "/ef/ef6/modeling/designer/advanced/edmx/msl-spec",
```
```suggestion
            "redirect_url": "/ef/ef6/modeling/designer/advanced/edmx/msl-spec"
```

mairaw commented on .openpublishing.redirection.json:
```diff
@@ -765,6 +765,26 @@
+        },
+        {
+            "source_path": "docs/framework/data/adonet/ef/language-reference/ssdl-specification.md",
+            "redirect_url": "/ef/ef6/modeling/designer/advanced/edmx/ssdl-spec",
```
```suggestion
            "redirect_url": "/ef/ef6/modeling/designer/advanced/edmx/ssdl-spec"
```

mairaw commented on .openpublishing.redirection.json:
```diff
@@ -765,6 +765,26 @@
+        {
+            "source_path": "docs/framework/data/adonet/ef/language-reference/ssdl-specification.md",
+            "redirect_url": "/ef/ef6/modeling/designer/advanced/edmx/ssdl-spec",
+            "redirect_document_id": true
```
```suggestion
```

mairaw reviewed (changes requested):
I missed this last time. We shouldn't have redirect_document_id for these since those articles already exist and have their own BI for a while.

DennisLee-DennisLee committed:
Removed the "redirect_document_id" dict value from new redirect entries.

mairaw commented:
there are some merge conflicts here now that I can't solve via UI. Can you take care of that @DennisLee-DennisLee?

DennisLee-DennisLee committed:
Fixed merge conflict (will delete docs/framework/data/adonet/ef/language-reference/msl-specification.md).

mairaw reviewed (approved):
LGTM 🚀
